sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClient.java

```diff
@@ -157,7 +157,7 @@
     public SyncPoller<FormRecognizerOperationResult, List<RecognizedForm>>
         beginRecognizeCustomForms(String modelId, InputStream form, long length,
         RecognizeCustomFormsOptions recognizeCustomFormsOptions, Context context) {
-        Flux<ByteBuffer> buffer = toFluxByteBuffer(form);
+        Flux<ByteBuffer> buffer = toFluxByteBuffer(form).cache();
```

Comment:
You'll want to use `.cache().map(ByteBuffer::duplicate)` here. Right now, this will cache the list of `ByteBuffer`s and replay them every time the `Flux` is re-subscribed. But the issue is that the same `ByteBuffer`s are emitted, this results in an issue where the first emission of the `ByteBuffer` reads it to completion (effectively `hasRemaining() == false`). So, subsequent subscriptions will attempt to read the buffer again but will result in no data. `ByteBuffer.duplicate` creates another view on top of the underlying data, in this case it is a `byte[]` as these are `HeapByteBuffer`s, and this new view will enable each re-subscription to have full access to the underlying data.

Reply:
I am concerned if adding `.cache().ByteBuffer.asReadOnlyBuffer` are making the API's hard to use and we are relying too much on the users to have a deeper understanding of the Flux data provided and reactive programming. 
Thus making it hard to use and un-approachable. 
@alzimmermsft Thoughts on continuing to use `toFluxByteBuffer` providing in FormRecognizer?

Reply:
Given that the current converter is only in implementation I think we still should move to the API in `FluxUtil`. As a follow-up we can add another overload to the `FluxUtil` to take a flag to make the produced `Flux<ByteBuffer>` replay-able.